rootfs/etc/nginx/lua/configuration.lua

```diff
@@ -29,13 +32,62 @@ local function fetch_request_body()
   return body
 end
 
+function _M.get_pem_cert_key(hostname)
+  return certificate_data:get(hostname)
+end
+
+local function handle_servers()
+  if ngx.var.request_method ~= "POST" then
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    ngx.print("Only POST requests are allowed!")
+    return
+  end
+
+  local raw_servers = fetch_request_body()
+
+  local ok, servers = pcall(json.decode, raw_servers)
+  if not ok then
+    ngx.log(ngx.ERR,  "could not parse servers: " .. tostring(servers))
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    return
+  end
+
+  local err_buf = {}
+  -- Update certificates and private keys for each host
+  for _, server in ipairs(servers) do
+    if server.hostname and server.sslCert.pemCertKey then
+      local success, err = certificate_data:set(server.hostname, server.sslCert.pemCertKey)
```

Comment:
When the dictionary is full Nginx will remove least used item to spare some space for the new one and try to fit the new item. In that case `success` will still be `true`. But IMO this is unacceptable for what we are trying to do. That would mean we delete the certificate of app1 so that we can make app2 work.

I think we should use `safe_set` (https://github.com/openresty/lua-nginx-module#ngxshareddictsafe_set) and when the error message is `"no memory"` immediately short-circuit from the loop and return 502, Internal Server Error and log the message.